contracts/ExecutionNodeVersionBeacon.cdc

```diff
@@ -0,0 +1,425 @@
+                UFix64(newUpdateBufferInBlocks) <= UFix64(ExecutionNodeVersionBeacon.versionUpdateBuffer) * (1.0 + ExecutionNodeVersionBeacon.versionUpdateBufferVariance),
+                message: "Can only change versionUpdateBuffer by allowed variance from existing buffer."
+            )
+
+            // Get current block height & clean up upcoming block boundaries
+            let currentBlockHeight = getCurrentBlock().height
+            ExecutionNodeVersionBeacon.archiveOldBlockBoundaries()
+
+            // No boundaries defined beyond current block, safe to make changes
+            if ExecutionNodeVersionBeacon.upcomingBlockBoundaries.length == 0 {
+                ExecutionNodeVersionBeacon.versionUpdateBuffer = newUpdateBufferInBlocks
+            } else {
+                // Get the proximal upcoming boundary
+                let nextBlockBoundary = ExecutionNodeVersionBeacon.upcomingBlockBoundaries[0]
+
+                // Ensure that the we're not currently within the old or new buffer period
+                // of the next block height boundary
+                assert(
+                    currentBlockHeight + ExecutionNodeVersionBeacon.versionUpdateBuffer < nextBlockBoundary &&
+                    currentBlockHeight + newUpdateBufferInBlocks < nextBlockBoundary,
+                    message: "Updating buffer now breaks version boundary update expectations. Try updating buffer after next version boundary."
+                )
+
+                // We can now safely change the versionUpdateBuffer
+                ExecutionNodeVersionBeacon.versionUpdateBuffer = newUpdateBufferInBlocks
+            }
+
+            emit ExecutionNodeVersionUpdateBufferChanged(newVersionUpdateBuffer: newUpdateBufferInBlocks)
+        }
+
+        pub fun changeVersionUpdateBufferVariance(_ newUpdateBufferVariance: UFix64) {
+            ExecutionNodeVersionBeacon.versionUpdateBufferVariance = newUpdateBufferVariance
+            emit ExecutionNodeVersionUpdateBufferVarianceChanged(newVersionUpdateBufferVariance: newUpdateBufferVariance)
+
+        }
+    }
+
+    /// Returns the current updateBuffer period within which Execution Nodes
+    /// can be assured the version will not change
+    pub fun getVersionUpdateBuffer(): UInt64 {
+        return self.versionUpdateBuffer
+    }
+
+    /// Returns the current updateBuffer period within which Execution Nodes
+    /// can be assured the version will not change
+    pub fun getVersionUpdateBufferVariance(): UFix64 {
+        return self.versionUpdateBufferVariance
+    }
+
+    /// Returns a copy of the full historical versionTable
+    pub fun getVersionTable(): {UInt64: Semver} {
+        return self.versionTable
+    }
+
+    /// Function that returns that version that was defined at the most
+    /// recent block height boundary
+    pub fun getCurrentExecutionNodeVersion(): Semver? {
+        pre {
+            self.versionTable.length > 0: "No current minimum version defined."
+        }
```

Comment:
I'd probably prefer to just return `nil` if there is an error here. no need to panic during a getter function